opc/resource_lbaas_policy.go

```diff
@@ -0,0 +1,901 @@
+						},
+					},
+				},
+			},
+			"set_request_header_policy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"header_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"action_when_header_exists": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"NOOP",
+								"PREPEND",
+								"APPEND",
+								"OVERWRITE",
+								"CLEAR",
+							}, true),
+						},
+						"action_when_header_value_is": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"action_when_header_value_is_not": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"ssl_negotiation_policy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"port": {
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: true,
+						},
+						"server_order_preference": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"Enabled",
+								"Disabled",
+							}, true),
+						},
+						"ssl_protocol": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"ssl_ciphers": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+			"trusted_certificate_policy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"trusted_certificate": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			// Read only attributes
+			"state": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourcePolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	lbaasClient, err := meta.(*Client).getLBaaSClient()
+	if err != nil {
+		return err
+	}
+	policyClient := lbaasClient.PolicyClient()
+
+	var lb lbaas.LoadBalancerContext
+	if loadBalancer, ok := d.GetOk("load_balancer"); ok {
+		lb = getLoadBalancerContextFromID(loadBalancer.(string))
+	}
+
+	input := lbaas.CreatePolicyInput{
+		Name: d.Get("name").(string),
+	}
+
+	if _, ok := d.GetOk("application_cookie_stickiness_policy"); ok {
```

Comment:
Can you have multiple policies (ie; `application_cookie_stickiness_policy` and `cloudgate_policy`) within the same resource? Or is it only one type of policy per resource?

Reply:
it looks like it's one type per policy so I think we'll have to rethink this resource a little

Reply:
I'm wondering if we should have a resource per type which sounds incredibly cumbersome but a little more manageable for users. The other option is to add `ConflictsWith: []string{"every","type","other","than","this","one"}` which is easier but still awkward for the user

Reply:
Just one type per policy.  It is possible to change a resource definition from one type to another (add a test for this). However getting this to work required *a bit of a hack.* :-(

I explored several design approaches - setting the type as an attribute and validating all the different attribute combinations (closer fit to the API) didn't work to well, the a separate policy resource for each type, and the implemented mutually exclusive block per type. The current approach felt like the best compromise and maintains some consistency with the user experience seeing the Policies in the UI.  

We can revisiting making them separate resources if necessary.

Reply:
I think we will need to consider multiple resources here. My concern is that a user could expect a policy to change if they swap type blocks. We could write out a swap for it in the provider/sdk but I'm not sure that would be better than separate resources per type. Let's discuss more tomorrow